sdk/go/pulumi/run.go

```diff
@@ -38,15 +38,22 @@ type RunOption func(*RunInfo)
 // to register resources and orchestrate deployment activities.  This connects back to the Pulumi engine using gRPC.
 // If the program fails, the process will be terminated and the function will not return.
 func Run(body RunFunc, opts ...RunOption) {
-	if err := RunErr(body, opts...); err != nil {
-		if err != ErrPlugins {
-			fmt.Fprintf(os.Stderr, "error: program failed: %v\n", err)
-			os.Exit(1)
-		}
+	err := RunErr(body, opts...)
+	if err == nil {
+		return
+	}
 
+	if err == ErrPlugins {
 		printRequiredPlugins()
 		os.Exit(0)
 	}
+
+	// Log the error message
+	if ctx, e := NewContext(context.TODO(), getEnvInfo()); e == nil {
```

Comment:
Why is this code trying to recover from err in `NewContext` response? Is this something actually expected to happen? Perhaps `contract.IgnoreError` on it or even `contract.AssertNoError` - it would seem it should never really happen.

Reply:
Well the code is still swallowing the err, so it handles it. This seems to be not great, because there is no explanation why this err should be swallowed and not treated normally.